scheduling-manager/src/main/java/com/amazonaws/blox/scheduling/state/PaginatedResponseIterator.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://aws.amazon.com/apache2.0/
+ *
+ * or in the "LICENSE" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+package com.amazonaws.blox.scheduling.state;
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.Spliterator;
+import java.util.Spliterators;
+import java.util.function.Function;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+import lombok.RequiredArgsConstructor;
+
+/**
+ * Abstract iterator that exposes a paginated list API call as an Iterator/Stream.
+ *
+ * @param <T> The response type returned by the list API call
+ */
+@RequiredArgsConstructor
+class PaginatedResponseIterator<T> implements Iterator<T> {
```

Comment:
why are these classes not public?

Reply:
Think it was left over from extracting this as a private inner class on one of the listers. Should it be public? I do not have strong feelings about java class visibility beyond what's needed to make the compiler happy, so I'll defer to whatever your guideline is here.